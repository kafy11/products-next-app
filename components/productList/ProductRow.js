import LinkNext from '../LinkNext';
import styles from '../../styles/ProductRow.module.css';

const ProductRow = ({ product, onDelete }) => {
    const { product_id, name, price, quantity } = product;

    return (
        <tr>
            <td>
                <i 
                    className={`fas fa-trash ${styles.btnDelete} product-row__delete-button`}
                    onClick={() => onDelete(product_id)}
                ></i>
            </td>
            <td>
                <LinkNext
                    href={`/product/${product_id}`}
                >
                    {name}
                </LinkNext>
            </td>
            <td>${price}.99</td>
            <td>{quantity}</td>
        </tr>
    );
}

export default ProductRow;