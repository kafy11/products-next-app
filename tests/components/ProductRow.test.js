import { shallow } from 'enzyme';
import ProductRow from '../../components/productList/ProductRow';
import products from '../fixtures/products';

let handleDelete, wrapper;
const product = products[0];

beforeEach(() => {
    handleDelete = jest.fn();
    wrapper = shallow(
        <ProductRow 
            product={product}
            onDelete={handleDelete} 
        />
    );
});

describe('ProductRow', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should handle onDelete', () => {
        wrapper.find('.product-row__delete-button').prop('onClick')();
        expect(handleDelete).toHaveBeenLastCalledWith(product.product_id);
    });
});