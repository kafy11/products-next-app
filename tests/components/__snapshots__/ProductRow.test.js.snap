// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow should render correctly 1`] = `
<tr>
  <td>
    <i
      className="fas fa-trash btnDelete product-row__delete-button"
      onClick={[Function]}
    />
  </td>
  <td>
    <LinkNext
      href="/product/1"
    >
      PS4
    </LinkNext>
  </td>
  <td>
    $
    399
    .99
  </td>
  <td>
    1
  </td>
</tr>
`;
